### PR TITLE
Create auth applications

### PR DESCRIPTION
**Pull request by davidw**
https://www.pivotaltracker.com/n/projects/252805/stories/105895576

### CODE REVIEW

**Comment by davidw** [comment]
@quady comments, suggestions?

**Comment by quady** [comment]
Does this also set the G5_AUTH_CLIENT_ID and G5_AUTH_CLIENT_SECRET?

**Comment by davidw** [comment]
@quady nope - should it?

**Comment by quady** [comment]
I know that is what it will need to do so Pro Services does not have to go into auth/heroku to set that up for each app. Currently those are the two variables we do not have a way to get from auth automatically.

**Comment by davidw** [comment]
@quady now it should have the ID and SECRET

File: lib/client_app_provision_modules.rb [hunk 1]
```diff
@@ -195,5 +195,44 @@ def provision
...
+                                                      redirect_uri: client_app.g5_auth_redirect_uri, superapp: true},
+                                        access_token: access_token}
+                               })
+      # Now set the variables up in the Heroku instance.
+      GithubHerokuDeployer.heroku_config_set(
+        {'G5_AUTH_ENDPOINT' => client_app.g5_auth_endpoint,
+         'G5_AUTH_REDIRECT_URI' => client_app.g5_auth_redirect_uri,
+         'G5_AUTH_USERNAME' => ENV['G5_AUTH_USERNAME'],
+         'G5_AUTH_PASSWORD' => ENV['G5_AUTH_PASSWORD'],
+         'G5_AUTH_CLIENT_ID' => response.parsed_response['uid'],
```
Comment (quady) [inline lib/client_app_provision_modules.rb]: Awesome, just want to make sure that this is the auth app client id, which should a md5 or sha string, something like that, not a url formatted uid.

If so, :+1:

Comment (davidw) [inline lib/client_app_provision_modules.rb]: Yes, it's the 'uid', which is big long string like `"b560029f4bd70a7ce02f3dcea84b24d61a6b4cca75195d415fa961556683b645"`

File: lib/client_app_provision_modules.rb [hunk 2]
```diff
@@ -195,5 +195,44 @@ def provision
...
+      @client_app = client_app
+      @defaults = DEFAULTS["deploys"]["initial"][client_app.app_type]
+      @http_client = HTTParty
+    end
+
+    def provision
+      Rails.logger.info "Setting up auth for #{client_app.name}..."
+      # Get an access token from the auth system.
+      authclient =
+        G5AuthenticationClient::Client.new(:username => ENV['G5_AUTH_USERNAME'],
```
Comment (dpetersen) [inline lib/client_app_provision_modules.rb]: I'm 90% sure there's a way to get these variables used automatically, but I'm having a hard time tracking it down. Something to do with this: https://github.com/G5/g5_authentication_client/blob/f8e5f96486ca8b25b16aea1a75fae0757b244505/lib/g5_authentication_client/configuration.rb

**Comment by dpetersen** [comment]
Seems legit. This will be nice. :+1:

**Pull request merged**